packages/@uppy/aws-s3/src/index.js

```diff
@@ -135,13 +135,23 @@ module.exports = class AwsS3 extends Plugin {
     const valid = typeof params === 'object' && params &&
       typeof params.url === 'string' &&
       (typeof params.fields === 'object' || params.fields == null) &&
-      (params.method == null || /^(put|post)$/i.test(params.method))
+      params.method == null;
 
     if (!valid) {
       const err = new TypeError(`AwsS3: got incorrect result from 'getUploadParameters()' for file '${file.name}', expected an object '{ url, method, fields, headers }' but got '${JSON.stringify(params)}' instead.\nSee https://uppy.io/docs/aws-s3/#getUploadParameters-file for more on the expected format.`)
       console.error(err)
       throw err
     }
+
+    const methodIsValid = /^(put|post)$/i.test(params.method);
```

Comment:
this will fail if `params.method` is `null`. I think it should be:

```suggestion
    const methodIsValid = params.method == null || /^(put|post)$/i.test(params.method)
```